### PR TITLE
[4.4] Prevent step target to be recorded with empty spaces and prevent tour run fails

### PR DESCRIPTION
**Pull request by obuisard**
### Summary of Changes

The step target is trimmed to avoid javascript errors and preventing a tour to start (with an error message that does not really explain what was wrong)

### Testing Instructions

Edit a tour step and edit the target, adding spaces before and after the content of the field.
Run the tour you have edited the step from.
The tour MAY fail (depending on the browser/system used). It failed on Safari.
Apply the patch.
Edit the target field for the step (so it can be saved). Once saved, the step target field should not contain any space.

### Actual result BEFORE applying this Pull Request

The tour may not start (an error message is shown when you start the tour).

### Expected result AFTER applying this Pull Request

The tour runs.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Table/StepTable.php [hunk 1]
```diff
@@ -85,4 +85,28 @@ public function store($updateNulls = true)
...
         return parent::store($updateNulls);
     }
+
+    /**
+     * Overloaded check function
+     *
+     * @return  boolean  True on success, false on failure
+     *
+     * @see     \Joomla\CMS\Table\Table::check
+     * @since   1.5
```
Comment (Quy) [inline administrator/components/com_guidedtours/src/Table/StepTable.php]:
```suggestion
     * @since   __DEPLOY_VERSION__
```

Comment (obuisard) [reply administrator/components/com_guidedtours/src/Table/StepTable.php]: Thanks!

**Comment by exlemor** [comment]
@obuisard 

Tested this under Joomla 4.4.4 and under Safari v17.4.1 (19618.1.15.11.14) on Mac OS Sonoma 14.4.1 and I could not have it fail. [I have checked that there is a space before and after the field]. 

Sorry.

**Comment by obuisard** [comment]
> @obuisard 
> 
> Tested this under Joomla 4.4.4 and under Safari v17.4.1 (19618.1.15.11.14) on Mac OS Sonoma 14.4.1 and I could not have it fail. [I have checked that there is a space before and after the field]. 
> 
> Sorry.

Thank you, Emmanuel @exlemor

**Comment by fgsw** [comment]
Can't test Pull Request as tour not failed on

- MacOS 14.5
- Safari 17.5, Firefox 129
- Error Reporting: Maximum.

**Comment by obuisard** [comment]
I am going to close this as nobody has been able to reproduce.
Thank you all for testing :-)

**Pull request closed without merging**